controller/adr.rst

```diff
@@ -0,0 +1,135 @@
+.. index::
+    single: Action Domain Responder approach
+
+How to implement the ADR pattern
+================================
+
+In Symfony, you're used to implement the MVC pattern and extending the default :class:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller`
+class.
+Since the 3.3 update, Symfony is capable of using natively the ADR approach.
+
+Updating your configuration
+---------------------------
+
+As the framework evolve, you must update the services.yml file in order to
+use the latest features of the DependencyInjection component, this way, here's the updates::
+
+    # ...
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        # Allow to load every actions
+        AppBundle\Action\:
+            resource: '../../src/AppBundle/Action/'
+            tags: ['controller.service_arguments']
+            public: true
```

Comment:
This behavior has been modified for 3.3, now all the services tagged with `controller.service_arguments` are public by definition and according to https://github.com/symfony/symfony-docs/pull/8415 we can remove it so.